src/Units/UnitModel.java

```diff
@@ -131,6 +140,24 @@ private UnitModel(String pName, long pRole, int cost, int pAmmoMax, int pFuelMax
     }
   }
 
+  private void setCalculatedProps()
+  {
+    boolean isDirect = weapons.size() > 0;
+    boolean isIndirect = false;
+    for( WeaponModel wm : weapons )
+    {
+      if( wm.rangeMax == 1 )
+        continue;
+      isDirect = false;
+      isIndirect = true;
+      break;
+    }
+    if( isDirect )
+      this.role |= DIRECT;
+    if( isIndirect )
+      this.role |= INDIRECT;
+  }
```

Comment:
This is good. Not sure why we didn't have something like it before.

Reply:
I wanted to do silly things instead.